### ISSUE TITLE
vsoagent /unconfigure only allows you to unconfigure the last agent configured

### ISSUE THREAD

**Issue by rkeithhill** [Microsoft/vsts-agent#179]
That is unfortunate if I have four agents configured in the order Agent1, Agent2, Agent3 and Agent4 **and** I want to unconfigure Agent1.  Why doesn't `/unconfigure` take an agent name to unconfigure?  Also, why does the settings.json only show settings for the last configured agent?  I suspect these two issues are related.

**Comment by bryanmacfarlane** [comment]
You have the wrong agent.  vsoagent is the legacy windows agent not in this repo.

That said, unconfigure for this one is in PR.

The answer to your question is each agent is standalong and xcopy by design.  Agents don't manage each other.

**Comment by rkeithhill** [comment]
Thanks for the reply.  BTW - legacy?  But we're on TFS 2015 Update 1.  Does Update2 have a new agent implementation?  Keep in mind we use TFS on-prem.

**Comment by bryanmacfarlane** [comment]
Yeah.  In the 2015 product, we shipped a windows agent that was closed source and built with the TFS product.  It worked only on windows only.  For OSX and Linux, we created a node based agent: https://github.com/Microsoft/vso-agent - so, we're managing and fixing issues in two agents - not fun.  The .net is shipping net core cross platform so it's an opportunity for us to combine the two and have one agent.

The effort in this repo is a net core cross platform single agent that will combine the two of those and ship with the next major version of TFS.  

The code and issues in this repo are for that effort.

We are trying to deprecate the node agent in a week or two.  So, the drops in the releases here will work with TFS2015 for OSX and Linux.
